File: 백준/Bronze/9046. 복호화/복호화.py
import sys

input = sys.stdin.readline

t = int(input())
for _ in range(t):
    cnt= {}
    all_char = []
    words = input().split()
    for word in words:
        all_char.extend(list(word))

    for char in all_char:
        if char in cnt:
            cnt[char] += 1
        else:
            cnt[char] = 1

    max_f = max(cnt.values())
    most_chars = [char for char, count in cnt.items() if count == max_f]

    if len(most_chars) == 1:
        print(most_chars[0])
    else:
        print('?')

File: 백준/Bronze/9046. 복호화/복호화.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        for(int i=0;i<t;i++){
            String[] words = br.readLine().split(" ");
            Map<Character, Integer> charCnt = new HashMap<>();
            for(String word:words){
                for(char c: word.toCharArray()){
                    charCnt.put(c,charCnt.getOrDefault(c,0)+1);
                }
            }
            int maxF = Collections.max(charCnt.values());
            List<Character> mostF = new ArrayList<>();
            for(Map.Entry<Character,Integer> entry:charCnt.entrySet()){
                if(entry.getValue()==maxF){
                    mostF.add(entry.getKey());
                }
            }
            if(mostF.size()==1){
                System.out.println(mostF.get(0));
            }else{
                System.out.println("?");
            }
        }
    }
}